python/ray/data/_internal/datasource/parquet_datasink.py

```diff
@@ -36,6 +37,59 @@
 logger = logging.getLogger(__name__)
 
 
+def choose_row_group_limits(
+    row_group_size: Optional[int],
+    min_rows_per_file: Optional[int] = None,
+    max_rows_per_file: Optional[int] = None,
+) -> tuple[int, int, int]:
+    """
+    Decide the (min_rows_per_group, max_rows_per_group, max_rows_per_file) for each **row group** so that every *file*
+    ends up within the caller’s [min_rows_per_file, max_rows_per_file] window.
+
+    Returns
+    -------
+    (min_rows_per_group, max_rows_per_group, max_rows_per_file)
+    """
+
+    ### From Pyarrow docs:
+    # max_rows_per_file : int, default 0
+    # Maximum number of rows per file. If greater than 0 then this will
+    # limit how many rows are placed in any single file. Otherwise there
+    # will be no limit and one file will be created in each output
+    # directory unless files need to be closed to respect max_open_files
+    max_rows_per_file = max_rows_per_file or 0
+
+    lower = min_rows_per_file or 0
+    # the upper limit must be a positive number/non-zero
+    upper = min(
+        max_rows_per_file or ARROW_DEFAULT_MAX_GROUP_SIZE, ARROW_DEFAULT_MAX_GROUP_SIZE
+    )
+
+    def clamp(value: int) -> int:
+        """Clamp *value* to [lower, upper], ignoring zero (unbounded) limits,
+        then respect the Arrow hard cap."""
+        return max(lower, min(value, upper))
+
+    if row_group_size is None:
+        min_row_group_size, max_row_group_size = lower, upper
+    else:
+        # Fixed-sized groups (post clamping to the [lower, upper] range)
+        clamped_group_size = clamp(row_group_size)
+        min_row_group_size = max_row_group_size = clamped_group_size
+
+    # Never let the lower bound exceed the upper
+    if min_row_group_size > max_row_group_size:
```

Comment:
Warnings should be against the input data not transformed. Otherwise users would have no idea where these values came from